### PR TITLE
fix: include catalog_visibility filter in courseware_content

### PR DESCRIPTION
**Pull request by MoisesGSalas**
The catalog_visibility filter is passed to the `courseware_content` index when performing a search using the legacy student dashboard view. If not included an error like follows is raised:

    MeilisearchApiError. Error code: invalid_search_filter. Error message:
    Attribute `catalog_visibility` is not filterable

I didn't dive to deep on this, but seems like the filter is added by the [`LmsSearchFilterGenerator`](https://github.com/openedx/edx-platform/blob/master/lms/lib/courseware_search/lms_filter_generator.py#L63-L66).

The normal course search in the Learning MFE doesn't throw any errors because the additional filters are not added when the search is [narrowed to a single course](https://github.com/openedx/edx-platform/blob/9cade7a3488eee24109311cb145747a01348df14/lms/lib/courseware_search/lms_filter_generator.py#L53-L54).

I'm not quite sure if this is the right approach considering that `catalog_visibility` doesn't appear in the documents for `courseware_content`, so the filter wouldn't really work.

I also noticed that if you configure the setting `SEARCH_SKIP_INVITATION_ONLY_FILTERING=False` you get `Unknown value type: <class 'bool'>` in both the dashboard search and the catalog search, because [`get_filter_rule`](https://github.com/openedx/edx-search/blob/15f34fbcb24afd96821a3a4a66889f9a66ce03c3/search/meilisearch.py/#L423) doesn't handle the bool case, and if it did we probably would get the filter error too.

## Testing instructions

On a tutor environment:

1. Because meilisearch is not included in nightly, I cherry-picked the meilisearch commit into the [nightly branch](https://github.com/eduNEXT/tutor/commit/817977f56a46b71b309d2f1851ddea662d71be01).
2. Created a small plugin to disable the learner-dashboard MFE:
    ```python
    from tutormfe.hooks import MFE_APPS
    
    @MFE_APPS.add()
    def _remove_mfes(mfes):
        mfes.pop("learner-dashboard")
        return mfes
    
    ```
3. Created a test course with a single 'Text' component with a sample search term 'exampleterm'
4. Search for 'exampleterm' on the dashboard search, it will throw _'An error occurred when searching for "exampleterm"'_
5. Use a an `edx-search` version with the fix and run `./manage.py lms shell -c "import search.meilisearch; search.meilisearch.create_indexes()"`
6. Repeat 4, this time the search should work.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @MoisesGSalas!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.


#### :radio_button: Update the status of your PR

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.




#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/openedx-unmaintained`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

**Comment by pomegranited** [comment]
Hi @MoisesGSalas , thank you for this PR, and apologies for my late reply (I've been on leave). Will review it this week.

**Review by pomegranited — APPROVED** [review]
:+1: Thank you for this fix @MoisesGSalas ! One nit to fix and we can find someone with merge rights here to merge this.

I've run out of CC time this year to dig deeper into the bugs you raised with this index though, but feel free to create an issue on this repo if they're affecting your instances.

- [x] I tested this on my tutor nightly stack using the PR instructions
- [x] I read through the code
- [x] <strike>I checked for accessibility issues by using my keyboard to navigate</strike> N/A
- [x] Includes documentation -- one nit re that comment
- [x] <strike>User-facing strings are extracted for translation</strike>

File: search/meilisearch.py [hunk 1]
```diff
@@ -104,6 +104,7 @@
         PRIMARY_KEY_FIELD_NAME,  # exclude some specific documents based on ID
         "course",  # search courseware content by course
         "org",  # used during indexing
+        "catalog_visibility",  # exclude visibility="none"
```
[the remark this answers is not in the document]
Comment (pomegranited) [reply search/meilisearch.py:107]: FYI relevant PR was https://github.com/openedx/edx-platform/pull/28518

[the remark this answers is not in the document]
Comment (MoisesGSalas) [reply search/meilisearch.py]: Done

**Comment by MoisesGSalas** [comment]
Thanks @pomegranited, I have not experienced that error in any instance, I only noticed it while exploring the original issue with the dashboard search.

I tagged you because I saw you were the last one to work on this and for some reason I assumed you had write permissions. Can you guide me on who could be a good last reviewer?

**Comment by pomegranited** [comment]
@MoisesGSalas 

> Can you guide me on who could be a good last reviewer?

Sure thing -- @Ali-Salman29 will be [maintaining this repo](https://discuss.openedx.org/t/core-contributor-nomination-ali-salman/14339) soon. @Ali-Salman29 could you take a look at this PR for us?

**Comment by salman299** [comment]
@MoisesGSalas 

```
{
   "properties":{
      "content":{
         "properties":{
            "display_name":{
               "type":"text",
               "fields":{
                  "keyword":{
                     "type":"keyword",
                     "ignore_above":256
                  }
               }
            },
            "html_content":{
               "type":"text",
               "fields":{
                  "keyword":{
                     "type":"keyword",
                     "ignore_above":256
                  }
               }
            }
         }
      },
      "content_groups":{
         "type":"keyword"
      },
      "content_type":{
         "type":"keyword"
      },
      "course":{
         "type":"keyword"
      },
      "course_name":{
         "type":"keyword"
      },
      "id":{
         "type":"keyword"
      },
      "location":{
         "type":"keyword"
      },
      "org":{
         "type":"keyword"
      },
      "start_date":{
         "type":"date"
      }
   }
}
```

Both Elasticsearch and Meilisearch use the courseware_content index for this query. There are no mappings for courseware_index in the Elasticsearch mapping, but if the filter is passed, it ignores it.

For Meilisearch, the filter list must include all the queried filters. In this case, the `courseware_index` is passed but there wasn't any filter for it; therefore it was throwing an error.

It's recommended to add all other missing fields `content_groups`, `content_type`, and `location` to the filter list so that it doesn't throw an error when these values are passed in the query.

In conclusion, the PR looks good to me. I don't have writes to merge this PR but if anyone could help it would be useful.
@pomegranited

**Review by salman299 — APPROVED** [review]

**Comment by Ali-Salman29** [comment]
@MoisesGSalas 

```
{
   "properties":{
      "content":{
         "properties":{
            "display_name":{
               "type":"text",
               "fields":{
                  "keyword":{
                     "type":"keyword",
                     "ignore_above":256
                  }
               }
            },
            "html_content":{
               "type":"text",
               "fields":{
                  "keyword":{
                     "type":"keyword",
                     "ignore_above":256
                  }
               }
            }
         }
      },
      "content_groups":{
         "type":"keyword"
      },
      "content_type":{
         "type":"keyword"
      },
      "course":{
         "type":"keyword"
      },
      "course_name":{
         "type":"keyword"
      },
      "id":{
         "type":"keyword"
      },
      "location":{
         "type":"keyword"
      },
      "org":{
         "type":"keyword"
      },
      "start_date":{
         "type":"date"
      }
   }
}
```

Both Elasticsearch and Meilisearch use the courseware_content index for this query. There are no mappings for courseware_index in the Elasticsearch mapping, but if the filter is passed, it ignores it.

For Meilisearch, the filter list must include all the queried filters. In this case, the `courseware_index` is passed but there wasn't any filter for it; therefore it was throwing an error.

It's recommended to add all other missing fields `content_groups`, `content_type`, and `location` to the filter list so that it doesn't throw an error when these values are passed in the query.

In conclusion, the PR looks good to me. I don't have writes to merge this PR but if anyone could help it would be useful.
@pomegranited

**Review by Ali-Salman29 — APPROVED** [review]

**Comment by pomegranited** [comment]
Thank you for your reviews @Ali-Salman29 and @MoisesGSalas! I don't have permission to merge here -- could one of you handle this, and tag a new release for us?

**Comment by Ali-Salman29** [comment]
@MoisesGSalas Can you please do it for now? My onboarding is still in progress.

**Comment by MoisesGSalas** [comment]
@Ali-Salman29, sorry I can't, I don't have write permissions on this repository.

**Comment by Ali-Salman29** [comment]
@pomegranited, I got the write access. Should I merge the PR?

**Comment by pomegranited** [comment]
Oh great, yes please @Ali-Salman29 !
We'll also need a version bump to get this fix into the platform. If you can't do that on this branch, you can submit a new PR to handle it.

**Comment by Ali-Salman29** [comment]
Thank You @pomegranited. I'm merging the PR.
